common-services-api/src/main/java/org/pentaho/big/data/api/jdbc/impl/LazyDelegatingDriver.java

```diff
@@ -45,19 +45,23 @@
 public class LazyDelegatingDriver implements Driver {
   private final DriverLocatorImpl driverLocator;
   private final HasRegisterDriver hasRegisterDriver;
+  private final HasDeregisterDriver hasDeregisterDriver;
   private Driver delegate;
   private DelegatingDriver delegatingDriver;
 
   public LazyDelegatingDriver( DriverLocatorImpl driverLocator ) throws SQLException {
-    this( driverLocator, DriverManager::registerDriver );
+    this( driverLocator, DriverManager::registerDriver, DriverManager::deregisterDriver );
   }
 
-  public LazyDelegatingDriver( DriverLocatorImpl driverLocator, HasRegisterDriver hasRegisterDriver )
+  public LazyDelegatingDriver( DriverLocatorImpl driverLocator,
+                               HasRegisterDriver hasRegisterDriver, HasDeregisterDriver hasDeregisterDriver )
     throws SQLException {
     this.driverLocator = driverLocator;
     this.hasRegisterDriver = hasRegisterDriver;
+    this.hasDeregisterDriver = hasDeregisterDriver;
     this.delegatingDriver = new DelegatingDriver( this );
     hasRegisterDriver.registerDriver( delegatingDriver );
+    hasDeregisterDriver.deregisterDriver( delegatingDriver );
```

Comment:
If deregistering it immediately, why register it in the first place?

I'm not aware of the context of this code, so I'm not sure the driver needs to be registered to perform its job (but I would guess so, as the description of `org.pentaho.di.core.database.DelegatingDriver` says _"This delegating driver allows the plugin system to be used in a separate class loader while making DriverManager believe that the parent class loader is used."_).

So I would call `hasDeregisterDriver.deregisterDriver` in a `destroy` method like in `ClusterInitializingDriver` (but who creates them would have to hold references and call it when appropriate).

Reply:
I think the hasDeregisterDriver is a functional call.
```
  public LazyDelegatingDriver( DriverLocatorImpl driverLocator ) throws SQLException {
    this( driverLocator, DriverManager::registerDriver, DriverManager::deregisterDriver );
  }
```
  I'll test this out some more to confirm we can still use the Driver.  I agree with you; if it's deregistering right away it would be pointless and break functionality.